pvlib/pvsystem.py

```diff
@@ -1473,6 +1449,83 @@ def get_iam(self, aoi, iam_model='physical'):
         else:
             raise ValueError(model + ' is not a valid IAM model')
 
+    def get_cell_temperature(self, poa_global, temp_air, wind_speed, model,
+                             **kwargs):
+        """
+        Determine cell temperature using the method specified by ``model``.
+
+        Parameters
+        ----------
+        poa_global : numeric
+            Total incident irradiance [W/m^2]
+
+        temp_air : numeric
+            Ambient dry bulb temperature [C]
+
+        wind_speed : numeric
+            Wind speed [m/s]
+
+        model : str
+            Supported models include ``'sapm'``, ``'pvsyst'``,
+            ``'faiman'``, ``'fuentes'``, and ``'noct_sam'``
+
+        **kwargs
+            Extra arguments passed to the model function.
+
+        Returns
+        -------
+        numeric
+            values in degrees C.
+
+        Notes
+        -----
+        Some temperature models have requirements for the input types;
+        see the documentation of the underlying model function for details.
+        """
+        if model == 'sapm':
+            func = temperature.sapm_cell
+            params = _build_kwargs(['a', 'b', 'deltaT'],
+                                   self.temperature_model_parameters)
+        elif model == 'pvsyst':
+            func = temperature.pvsyst_cell
+            params = {
+                **_build_kwargs(['eta_m', 'alpha_absorption'],
+                                self.module_parameters),
+                **_build_kwargs(['u_c', 'u_v'],
+                                self.temperature_model_parameters)
+            }
+        elif model == 'faiman':
+            func = temperature.faiman
+            params = _build_kwargs(['u0', 'u1'],
+                                   self.temperature_model_parameters)
+        elif model == 'fuentes':
+            func = temperature.fuentes
+            # default to using the Array attribute, but allow user to override
+            # with a custom surface_tilt value in temperature_model_parameters
+            params = _build_kwargs([
+                'noct_installed', 'module_height', 'wind_height', 'emissivity',
+                'absorption', 'surface_tilt', 'module_width', 'module_length'],
+                self.temperature_model_parameters)
+            if 'surface_tilt' not in params:
+                params['surface_tilt'] = self.surface_tilt
+        elif model == 'noct_sam':
+            func = temperature.noct_sam
+            params = _build_kwargs(['transmittance_absorptance', 'eta_m_ref',
+                                    'noct', 'array_height', 'mount_standoff'],
+                                   self.temperature_model_parameters)
+            if not {'noct', 'eta_m_ref'}.issubset(params):
```

Comment:
I'm not sure how I feel about this.  Without a check like this, the error message isn't as helpful as it could be -- `TypeError: noct_sam() missing 1 required positional argument: 'noct'` makes no mention of `temperature_model_parameters`.  But I'm not really a fan of having all these lists of `pvlib.temperature` parameter names in `pvlib.pvsystem`.

What do you think about letting python raise the `TypeError` and we add a hint to the error message like this:

```python
try:
    temperature_cell = func(poa_global, temp_air, wind_speed, **params)
except TypeError as e:
    msg = e.args[0] + (
        ".  Check temperature_model_parameters to make sure "
        "all required parameters are specified."
    )
    raise TypeError(msg)
```

Downside is that it adds irrelevant noise to TypeErrors caused by other problems.

Reply:
I won't stand in the way of this PR over error messaging. If we want to verify parameters at this stage, as a kindness to users, maybe a dict in `pvlib.temperature` akin to `pvlib.pvsystem._DC_MODEL_PARAMS`.

Reply:
I think the problem is that we're using `_build_kwargs` to build all of the arguments instead of just the optional keyword arguments. I suggest a pattern that performs explicit look ups instead of deferred checks. Fine with me to use the standard python KeyError or break into the kinder message template.

```python
args = (self.temperature_model_par[arg] for arg in required_args)
kwargs = _build_kwargs(optional_args, self.temperature_model_parameters) 
```

required_args and optional_args could come from a private dict like @cwhanse suggests. I'm also ok with hard coding it here (sounds easy). And I'm ok with using `inspect` (I vaguely recall considering this previously and thinking it was too complicated for pvlib, but I currently like it).

Reply:
I added something along these lines, let me know what you think.